cli/cli.js

```diff
@@ -72,38 +76,126 @@ const runYoCommand = (cmd, args, options, opts) => {
+            if (value.type === String) {
+                cmdString = `${cmdString} <value>`;
+            }
+            command.option(cmdString, value.description, value.default);
+        });
+    }
+
     command
         .allowUnknownOption()
         .description(opts.desc)
-        .action(args => {
-            const options = getCommandOptions(packageJson, process.argv.slice(2));
+        .action((first, second, third) => {
+            let args;
+            let cmdOptions;
+            let unknownArgs;
+            logger.debug(`first command arg: ${toString(first)}`);
+            logger.debug(`second command arg: ${toString(second)}`);
+            logger.debug(`third command arg: ${toString(third)}`);
+
+            if (opts.argument) {
+                // Option that contains arguments
+                // first=arguments second=cmdOptions third=unknownArgs
+                if (Array.isArray(first)) {
+                    // Var args unknown options are concatenated.
+                    // consider valid args before first unknown option (starts with -).
+                    args = [];
+                    unknownArgs = [];
+                    let unknown = false;
+                    first.forEach(item => {
+                        if (item.startsWith('-')) {
+                            unknown = true;
+                        }
+                        if (unknown) {
+                            unknownArgs.push(item);
+                        } else {
+                            args.push(item);
+                        }
+                    });
+                } else if (first !== undefined) {
+                    args = [first];
+                }
+                cmdOptions = second;
+                if (third) {
+                    unknownArgs = (unknownArgs || []).concat(third);
+                }
+            } else {
+                // Option that doesn't contain arguments
+                // first=cmdOptions second=unknownArgs
+                args = [];
+                cmdOptions = first;
+                unknownArgs = second || [];
+            }
+
+            const firstUnknownArg = Array.isArray(unknownArgs) && unknownArgs.length > 0 ? unknownArgs[0] : undefined;
+            if (key === 'app' && firstUnknownArg !== undefined && !firstUnknownArg.startsWith('-')) {
+                // Unknown commands.
+                const cmd = Object.values(unknownArgs).join('');
+                const availableCommands = program.commands.map(c => c._name);
+
+                const suggestion = didYouMean(cmd, availableCommands);
+                if (suggestion) {
+                    logger.info(`Did you mean ${chalk.yellow(suggestion)}?`);
+                }
+
+                logger.fatal(`${chalk.yellow(firstUnknownArg)} is not a known command. See '${chalk.white(`${CLI_NAME} --help`)}'.`);
+                return;
+            }
+
+            const customOptions = {};
```

Comment:
@mshima I have merged this for now, but honestly, I would like this part to be handled in a simpler way, by just adding a new flag for the command `--skip-sample-repo` and pass that flag in tests instead of doing this here so that both commands behave exactly the same, like an alias should be, can you do a new PR for that?

Reply:
This should be jhipster 7 only https://github.com/jhipster/generator-jhipster/pull/11770#issuecomment-634892631.
But I think we should remove `import-jdl` instead.

Reply:
That comment is about the import-jdl command being removed nit about handling tests, tests are internal and we can handle it however we like

Reply:
I understood @pascalgrimaud asked to keep import-jdl behavior unchanged for jhipster 6.

Reply:
Yes, I'd like to have the same current behavior for import-jdl.
I don't care if it can take a url instead of a file, as it would be an improvement. 
What I want is to keep:
jhipster import-jdl file.jdl : it imports all entities

Plz, don't change it now. No breaking change until v7.

Reply:
That behavior doesn't change and AFAIK there is no breaking change.

Reply:
thank you :)